### PR TITLE
feat(forge): export gas reports

### PR DESCRIPTION
**Pull request by brockelmore**
Adds the ability to export gas reports to `.gas-report` via cli input `--gas-report-export` or `--gre`

### CODE REVIEW

File: evm-adapters/src/gas_report.rs [hunk 1]
```diff
@@ -34,8 +35,8 @@ pub struct GasInfo {
 }
 
 impl GasReport {
-    pub fn new(report_for: Vec<String>) -> Self {
-        Self { report_for, ..Default::default() }
+    pub fn new(report_for: Vec<String>, export: bool) -> Self {
+        Self { report_for, stdout: !export, ..Default::default() }
```
Comment (mattsse) [inline evm-adapters/src/gas_report.rs:39]: should this rather be `colors`?

File: cli/src/cmd/test.rs [hunk 2]
```diff
@@ -438,7 +442,17 @@ fn test<A: ArtifactOutput + 'static>(
                 }
             }
             gas_report.finalize();
-            println!("{}", gas_report);
+            if gas_reporting_export {
+                let table_view = format!("{}", gas_report);
+                if let Err(e) = std::fs::write(".gas-report", table_view) {
+                    println!(
```
Comment (mattsse) [inline cli/src/cmd/test.rs:448]:
```suggestion
                    eprintln!(
```

**Review by mattsse — COMMENTED** [review]
`--gre` is essentially `> <output>` right?

**Comment by gakonst** [comment]
@transmissions11 what is the recommended way for moving forward with this?

**Comment by transmissions11** [comment]
imo we should remove forge test --gas-report and swap out the logic in forge snapshot to output these gas report tables into the .gas-snapshot file

i can probably do that with this PR as a base this week unless anyone wants to frontrun me

**Comment by gakonst** [comment]
Sounds great! Let's go with that then.

**Comment by mattsse** [comment]
supportive of that `snapshot` is basically an extension of `test` already so we just need to settle on the --gas-report format then?
we then also need some `--color` and/or `--theme` args I believe?

**Comment by gakonst** [comment]
Yeah printing to a file should be without colors!

**Comment by gakonst** [comment]
Closing in support of @transmissions11 integrating this functionality as part of `forge snapshot`

**Pull request closed without merging**